File: LeetCode/Easy/Question_1/1_TwoSum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:

        tmp = nums.copy()
        tmp.sort()

        i = 0
        j = len( tmp ) - 1

        while i < j :

            if tmp[i] + tmp[j] == target : break
            elif tmp[i] + tmp[j] < target : i+=1
            elif tmp[i] + tmp[j] > target : j-=1

        idx1 = nums.index(tmp[i])
        nums[idx1] = -1
        idx2 = nums.index(tmp[j])

        return [ idx1, idx2 ]

print( Solution().twoSum( [2,7,11,15], 9 ) )
print( Solution().twoSum( [3,2,4], 6 ) )
print( Solution().twoSum( [3,3], 6 ) )

# Beats 86.87% Runtime, 60ms
# Beats 92.17% Memory, 14.9mb

File: LeetCode/Easy/Question_1/1_TwoSum.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int* twoSum(int* nums, int numsSize, int target, int* returnSize){
    
    int* retArr = malloc( 2 * sizeof( int ) );

    for( int i = 0; i < numsSize; i++ )
    {
        for( int j = i+1; j < numsSize; j++ )
        {
            if( ( nums[i] + nums[j] ) == target )
            {                
                *returnSize = 2;
                retArr[0] = i;
                retArr[1] = j;

                return retArr;
            }
        }
    }

    *returnSize = 0;
    return retArr;
}

int main()
{
    int arr[3] = {1,2,3};
    int returnSize[2];

    int * retArr = twoSum( arr, 3, 5, returnSize );

    printf( "{%d, %d}\n", retArr[0], retArr[1] );

    return 0;
}

//  Beats 57.27% Runtime, 139ms
//  Beats 48.47% Memory, 6.5mb